### ISSUE TITLE
Soda2Consumer should not require username and password, specify token should be enough

### ISSUE THREAD

**Issue by phoenixjiangnan** [socrata/soda-java#23]
Why Soda2Consumer require username and password? Shouldn't providing token is enough? If I interact with the server via http, token is the only thing I need

**Comment by chrismetcalf** [comment]
You can use the constructor that just takes a base URL: https://github.com/socrata/soda-java/blob/270269d3eb4f4e64169801cbd3c2aade523bc2d9/src/main/java/com/socrata/api/Soda2Consumer.java#L28

Would that work for you instead?

**Comment by phoenixjiangnan** [comment]
Not really. According to section 'Throttling and Application Tokens' in https://dev.socrata.com/consumers/getting-started.html, if developers don't specify a token, there'll be throttling of app's requests.

**Comment by phoenixjiangnan** [comment]
Are you ok if I submit a PR to add another constructor, which only requires url and token, to Soda2Consumer?

**Comment by chrismetcalf** [comment]
@bowenli86 Sorry for not following up sooner, but of course a pull request would be welcome.

FYI, within the next day or two I'll be merging https://github.com/socrata/soda-java/pull/27, which should dramatically clean up running the tests and such, and will introduced TravisCI testing to our workflow as well. Long overdue.
